api-lib/HttpError.ts

```diff
@@ -14,45 +31,53 @@ export class PayloadTooLargeError extends Error implements HttpError {
   httpStatus = 413;
 }
 
-export class NotFoundError extends Error {
+export class NotFoundError extends Error implements HttpError {
   httpStatus = 404;
 }
 
-export class ForbiddenError extends Error {
+export class ForbiddenError extends Error implements HttpError {
   httpStatus = 403;
 }
 
-export class BadRequestError extends Error {
+export class BadRequestError extends Error implements HttpError {
   httpStatus = 400;
 }
 
-export function errorResponse(res: VercelResponse, error: any): VercelResponse {
+export class UnprocessableError extends Error implements HttpError {
+  httpStatus = 422;
+  details?: any;
+}
+
+export function errorResponse(res: VercelResponse, error: any): void {
   const statusCode = error.httpStatus || 500;
-  return errorResponseWithStatusCode(res, error, statusCode);
+  errorResponseWithStatusCode(res, error, statusCode);
 }
 
 export function errorResponseWithStatusCode(
   res: VercelResponse,
   error: any,
   statusCode: number
-): VercelResponse {
-  return res.status(statusCode).json({
-    message: error.message || 'Unexpected error',
-    // included at this level for backwards compat w/ older hasura, perhaps can remove
-    code: `${statusCode}`,
-    extensions: {
+): void {
+  Sentry.captureException(error);
+  // We have to await this flush otherwise we exit before reporting to sentry
+  Sentry.flush(awaitSentryFlushMs).then(() => {
```

Comment:
I think `Sentry.close` is probably more applicable given the lifecycle of serverless function invocations, but `flush` seems like a good conservative choice given that we don't have a total understanding of the inner-workings of vercel under the hood.